tests/unit/test_loaders.py

```diff
@@ -23,157 +23,216 @@
-        )
-        self.assertEqual(data['api_version'], '2012-10-01')
-
-    def test_load_service_model_version_not_found(self):
-        with self.assertRaises(ApiVersionNotFoundError):
-            data = self.loader.load_service_model(
-                'someservice',
-                api_version='2010-02-02'
-            )
-
-    def test_load_service_model_data_path_order(self):
-        # There's an s3/ directory both in our custom BOTO_DATA_PATH
-        # directory as well as in the botocore/data/ directory.
-        # Our path should win since the default built in path is always
-        # last.
-        data = self.loader.load_service_model('aws/s3')
-        self.assertTrue(data.get('WAS_OVERRIDEN_VIA_DATA_PATH'),
-                        "S3 model was loaded from botocore's default "
-                        "data path instead of from the BOTO_DATA_PATH"
-                        " directory.")
+            loader.load_data('baz')
 
+    @mock.patch('os.path.isdir', mock.Mock(return_value=True))
     def test_list_available_services(self):
-        avail = self.loader.list_available_services('')
-        self.assertEqual(sorted(avail), [
-            'aws',
-            'aws',
-            'someservice',
-            'sub',
-        ])
-
-        aws_avail = self.loader.list_available_services('aws')
-        self.assertTrue(len(aws_avail) > 10)
-        self.assertTrue('ec2' in aws_avail)
-
-    def test_load_data_overridden(self):
-        self.overrides_path = os.path.join(
-            os.path.dirname(__file__),
-            'data_overrides'
-        )
-        self.environ['BOTO_DATA_PATH'] = "{0}{1}{2}".format(
-            self.overrides_path,
-            os.pathsep,
-            self.data_path
-        )
-        loader = Loader(data_path=self.environ['BOTO_DATA_PATH'])
-        # This should load the data the first data it finds.
-        data = loader.load_service_model(
-            'someservice',
-            api_version='2012-10-01'
-        )
-        # An overridden key.
-        self.assertEqual(data['api_version'], '2012-10-01')
-        # A key unique to the base.
-        self.assertEqual(data['something-else'], 'another')
-        # Ensure a key present in other variants is not there.
-        self.assertTrue('Purpose' not in data)
-
-    @mock.patch('os.pathsep', ';')
-    def test_search_path_on_windows(self):
-        # On windows, the search path is separated by ';' chars.
-        self.environ['BOTO_DATA_PATH'] = 'c:\\path1;c:\\path2'
-        # The builtin botocore data path is added as the last element
-        # so we're only interested in checking the two that we've added.
-        loader = Loader(data_path=self.environ['BOTO_DATA_PATH'])
-        paths = loader.get_search_paths()[:-1]
-        self.assertEqual(paths, ['c:\\path1', 'c:\\path2'])
-
-
-if __name__ == "__main__":
-    unittest.main()
+        # Fake mapping of directory name
+        # to directory names.
+        fake_directories = {
```

Comment:
I think that I would have actually set up a temporary directory with tempdir so I would not have to mock out some of the methods in the os library. Then set the data path to this that temporary directory.

For me either way is fine, just suggesting a different way in order that may simplify those two tests.